modules/account-lib/src/coin/baseCoin/blsKeyPair.ts

```diff
@@ -0,0 +1,62 @@
+import * as BLS from '@chainsafe/bls';
+import * as Crypto from '../../utils/crypto';
+import { KeyPairOptions, ByteKeys, isPrivateKey } from './iface';
+import { BaseKeyPair } from './baseKeyPair';
+import { AddressFormat } from './enum';
+import { NotImplementedError } from './errors';
+
+/**
+ * Base class for BLS keypairs.
+ */
+export abstract class BlsKeyPair implements BaseKeyPair {
+  protected keyPair: ByteKeys;
+  protected source?: KeyPairOptions;
+
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param {KeyPairOptions} source Either a master seed, a private key (extended or raw), or a public key
+   *     (extended, compressed, or uncompressed)
+   */
+  protected constructor(source?: KeyPairOptions) {
+    if (!source) {
+      const blsKeypair = BLS.generateKeyPair();
+      this.keyPair = { prv: blsKeypair.privateKey().toBytes(), pub: blsKeypair.publicKey().toBytesCompressed() };
```

Comment:
Let's just store the full `BLS` object and do format conversions in getter functions. Best to keep full context in state for flexibility and to get some validation for free. For example current impl of `recordKeysFromPrivateKey` doesn't validate the prv, but if we create and store a bls key object using the library we will likely get validation from the library for free